server/py/services/api/initial_data.py

```diff
@@ -52,68 +54,130 @@
 
 
 def init_data(
-    from_scratch: bool = False, perform_migrations_if_needed: bool = False
+    perform_migrations_if_needed: bool = False,
 ) -> None:
-    logger.info("Initializing DB data")
+    mlrun.utils.logger.info("Initializing DB data")
+
+    engine = framework.db.sqldb.sql_session.get_engine()
+    url = engine.url
+    from_scratch = False
+    if not sqlalchemy_utils.database_exists(url):
+        mlrun.utils.logger.info(
+            "Database does not exist, initializing from scratch",
+            database_url=url,
+        )
+        sqlalchemy_utils.create_database(url)
+        from_scratch = True
+    else:
+        has_tables = bool(sqlalchemy.inspect(engine).get_table_names())
+        if not has_tables:
+            mlrun.utils.logger.info(
+                "No tables found in the database, initializing from scratch",
+                database_url=url,
+            )
+            from_scratch = True
+
+    if from_scratch:
+        initialize_db_from_scratch(engine, url)
+    else:
+        _migrate_existing_data(
+            perform_migrations_if_needed,
+        )
+
+    mlrun.utils.logger.info("Initial data created")
+
+
+def create_schema(
```

Comment:
```suggestion
def _create_schema(
```

do it while they are not called externally